src/app/cube/details/action-panel/action-panel.component.ts

```diff
@@ -119,8 +62,8 @@ export class DetailsComponent implements OnInit, OnDestroy {
 
     try {
       const val = await this.cartService.addToCart(
-        this.author,
-        this.learningObjectName
+        this.learningObject.author.username,
```

Comment:
Since these details were all contained within the learningObject, I favored referencing them this way instead of having extra instance variables to store the author and learningObjectName.